field.go

```diff
@@ -175,7 +175,8 @@ func Stringer(key string, val fmt.Stringer) zapcore.Field {
 // Time constructs a zapcore.Field with the given key and value. The encoder
 // controls how the time is serialized.
 func Time(key string, val time.Time) zapcore.Field {
-	return zapcore.Field{Key: key, Type: zapcore.TimeType, Integer: val.UnixNano()}
+	// Use Interface instead of Integer here to keep location in time.
+	return zapcore.Field{Key: key, Type: zapcore.TimeType, Interface: val}
```

Comment:
I made another try: https://github.com/uber-go/zap/pull/425/commits/384c2f1fd8cda0b7df2a8a1762831400b74f5377
This time, we don't have extra allocate.